pkg/authentication/middlewares.go

```diff
@@ -40,12 +40,8 @@ func (m *Middleware) SetAllowListedEndpoints(endpointsPrefixes ...string) {
 
 func (m *Middleware) isAllowListed(r *http.Request) bool {
 	endpoint := r.URL.Path
-	for prefix, _ := range m.allowListedEndpoints {
-		if strings.HasPrefix(endpoint, prefix) {
-			return true
-		}
-	}
-	return false
+	_, ok := m.allowListedEndpoints[endpoint]
```

Comment:
what happens in the case of the slice being not initialized?does `range` deal with the nil pointer check?if not happy to swap otherwise we might need to check for the remote edge case

Reply:
This cannot happen since the construction funtion takes care of initializing the map. Plus, with `isAllowListed` being a "private" method we should be good to go.
But I'm open to refactor this

Reply:
nah that's good enough for me then